app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -311,6 +311,11 @@ public function getCartDataJs($multipage = true)
             $success = Mage::getStoreConfig('payment/boltpay/success');
             $close = Mage::getStoreConfig('payment/boltpay/close');
 
+            /*
+             * If we enable the config option "Add Bolt Button everywhere" - we should check the quote items
+             * before Bolt Popup open.
+             */
+            $isEmptyQuote = (!($sessionQuote->getItemsCollection()->count())) ? 'true' : 'false';
```

Comment:
should we have this check earlier? For instance,`$this->createBoltOrder` at line 268 should not be called if cart is empty

Reply:
@daisy1754 
But it show when "replace.phtml" is initializing:
app/code/community/Bolt/Boltpay/Helper/Data.php::canUseBolt()

Current method do not check the cart items number. So if we have enabled the option "Add the Bolt Button Everywhere" the script will init the button even we do not have any items in the cart. This issue does not appear if, as like in the Luma theme, the buttons do not appear, but if the checkout button does not hide so we have such issue. We see the Bolt Button.

Reply:
Just to clarify, I agree we need `$sessionQuote->getItemsCollection()->count()`

My proposal is move this check to above, around line 268. If cart is empty we should not call `$this->createBoltOrder` method

Reply:
now i understand, thanks, i will move it up